src/python/dxpy/__init__.py

```diff
@@ -609,6 +610,12 @@ def ensure_ascii(i):
 
                 response = pool_manager.request(_method, encoded_url, headers=_headers, body=body,
                                                 timeout=timeout, retries=False, **kwargs)
+                # Handle redirection manually for symlink files
+                if response.status in range(300, 399):
+                    redirect_url = response.headers.get('Location')
```

Comment:
what is expected result when `status in range(300, 399)`, but `Location` header is not provided? so we know we should redirect, but don't know where...

for now, it will throw `AssertionError('Should never reach this line: should never break out of loop')` and it is not retried...can you please raise some better exception? I would say from here so it gets retried, because I'd expect this is faulty behaviour and might be retriable...